core/amber/src/main/scala/edu/uci/ics/texera/web/SubscriptionManager.scala

```diff
@@ -0,0 +1,20 @@
+package edu.uci.ics.texera.web
+
+import rx.lang.scala.Subscription
+
+import scala.collection.mutable
+
+trait SubscriptionManager {
+
+  private val subscriptions = mutable.ArrayBuffer[Subscription]()
+
+  def addSubscription(sub: Subscription): Unit = {
+    subscriptions.append(sub)
+  }
+
+  def unsubscribeAll(): Unit = {
+    subscriptions.foreach(_.unsubscribe())
+    subscriptions.clear()
+  }
+
+}
```

Comment:
`rxjs`'s subscribable or subject can be used for multiple subscriptions. is it because we do not have the same api on the backend java/scala subscription?

Reply:
Although this can be replaced with rxjs subject, we currently only need addSubscription and unsubscribeAll these 2 functionalities.